graphql/introspection/introspection.go

```diff
@@ -299,3 +301,128 @@ func AddIntrospectionToSchema(schema *graphql.Schema) {
 
 	schema.Query = isQuery
 }
+
+// Copied from https://github.com/graphql/graphiql/blob/master/src/utility/introspectionQueries.js
+const introspectionQuery = `
+	query IntrospectionQuery {
+		__schema {
+			queryType { name }
+			mutationType { name }
+			types {
+				...FullType
+			}
+			directives {
+				name
+				description
+				locations
+				args {
+					...InputValue
+				}
+			}
+		}
+	}
+	fragment FullType on __Type {
+		kind
+		name
+		description
+		fields(includeDeprecated: true) {
+			name
+			description
+			args {
+				...InputValue
+			}
+			type {
+				...TypeRef
+			}
+			isDeprecated
+			deprecationReason
+		}
+		inputFields {
+			...InputValue
+		}
+		interfaces {
+			...TypeRef
+		}
+		enumValues(includeDeprecated: true) {
+			name
+			description
+			isDeprecated
+			deprecationReason
+		}
+		possibleTypes {
+			...TypeRef
+		}
+	}
+	fragment InputValue on __InputValue {
+		name
+		description
+		type { ...TypeRef }
+		defaultValue
+	}
+	fragment TypeRef on __Type {
+		kind
+		name
+		ofType {
+			kind
+			name
+			ofType {
+				kind
+				name
+				ofType {
+					kind
+					name
+					ofType {
+						kind
+						name
+						ofType {
+							kind
+							name
+							ofType {
+								kind
+								name
+								ofType {
+									kind
+									name
+								}
+							}
+						}
+					}
+				}
+			}
+		}
+	}
+`
+
+// ExecuteIntrospectionQueryOnSchema returns the result of executing a GraphQL
+// introspection query.
+//
+// Example:
+//
+// ```
+//	value, err := introspection.ExecuteIntrospectionQueryOnSchema(*schema)
+//	if err != nil {
+//		panic(err)
+//	}
+//
+//	valueJSON, err := json.Marshal(value)
+//	if err != nil {
+//		panic(err)
+//	}
+//
+//	fmt.Print(string(valueJSON))
+// ```
+func ExecuteIntrospectionQueryOnSchema(schema graphql.Schema) (interface{}, error) {
+	AddIntrospectionToSchema(&schema)
```

Comment:
Easy thing to do is to take `schemabuilder.Schema` and create a new `graphql.Schema` inside this function. This way the behavior won't be surprising (we don't change the parameter we're given).